E0084 needs to be updated to new format

From: src/test/compile-fail/E0084.rs

E0084 needs a span_label, updating it from:

```
error[E0084]: unsupported representation for zero-variant enum
  --> src/test/compile-fail/E0084.rs:12:1
   |
12 | enum Foo {} //~ ERROR E0084
   | ^^^^^^^^^^^
```

To:

```
error[E0084]: unsupported representation for zero-variant enum
  --> src/test/compile-fail/E0084.rs:12:1
   |
12 | enum Foo {} //~ ERROR E0084
   | ^^^^^^^^^^^ unsupported enum representation
```

--- a/src/librustc_typeck/check/mod.rs
+++ b/src/librustc_typeck/check/mod.rs
@@ -1245,8 +1245,11 @@ pub fn check_enum_variants<'a,'tcx>(ccx: &CrateCtxt<'a,'tcx>,
     let hint = *ccx.tcx.lookup_repr_hints(def_id).get(0).unwrap_or(&attr::ReprAny);
 
     if hint != attr::ReprAny && vs.is_empty() {
-        span_err!(ccx.tcx.sess, sp, E0084,
-            "unsupported representation for zero-variant enum");
+        struct_span_err!(
+            ccx.tcx.sess, sp, E0084,
+            "unsupported representation for zero-variant enum")
+            .span_label(sp, &format!("unsupported enum representation"))
+            .emit();
     }
 
     let repr_type_ty = ccx.tcx.enum_repr_type(Some(&hint)).to_ty(ccx.tcx);

--- a/src/test/compile-fail/E0084.rs
+++ b/src/test/compile-fail/E0084.rs
@@ -9,7 +9,9 @@
 // except according to those terms.
 
 #[repr(i32)]
-enum Foo {} //~ ERROR E0084
+enum Foo {}
+//~^ ERROR E0084
+//~| unsupported enum representation
 
 fn main() {
 }

